var/spack/repos/builtin/packages/py-petsc4py/package.py

```diff
@@ -28,20 +28,23 @@ class PyPetsc4py(PythonPackage):
     version('3.8.0',  sha256='b9b728e39245213cd8e74cf4724be9bb48bd295f99634135e37dbbdbec275244')
     version('3.7.0',  sha256='fb78b50c596c3ba6a097751dd9a379e7acaf57edd36311a3afa94caa4312ee08')
 
+    variant('mpi', default=True,  description='Activates MPI support')
+
     depends_on('py-cython', type='build', when='@develop')
     depends_on('python@2.6:2.8,3.3:', type=('build', 'run'))
     depends_on('py-setuptools', type='build')
     depends_on('py-numpy', type=('build', 'run'))
-    depends_on('py-mpi4py', type=('build', 'run'))
-
-    depends_on('petsc+mpi')
-    depends_on('petsc@develop+mpi', when='@develop')
-    depends_on('petsc@3.13:3.13.99+mpi', when='@3.13:3.13.99')
-    depends_on('petsc@3.12:3.12.99+mpi', when='@3.12:3.12.99')
-    depends_on('petsc@3.11:3.11.99+mpi', when='@3.11:3.11.99')
-    depends_on('petsc@3.10.3:3.10.99+mpi', when='@3.10.1:3.10.99')
-    depends_on('petsc@3.10:3.10.2+mpi', when='@3.10.0')
-    depends_on('petsc@3.9:3.9.99+mpi', when='@3.9:3.9.99')
-    depends_on('petsc@3.8:3.8.99+mpi', when='@3.8:3.8.99')
-    depends_on('petsc@3.7:3.7.99+mpi', when='@3.7:3.7.99')
-    depends_on('petsc@3.6:3.6.99+mpi', when='@3.6:3.6.99')
+    depends_on('py-mpi4py', when='+mpi', type=('build', 'run'))
+
+    depends_on('petsc+mpi', when='+mpi')
+    depends_on('petsc~mpi~hdf5~hypre~superlu-dist', when='~mpi')
```

Comment:
```suggestion
    depends_on('petsc~mpi', when='~mpi')
```
Maybe even:
```suggestion
    depends_on('petsc')
```
so it doesn't care whether or not it was built with MPI support.

Reply:
I'm happy to do what makes sense. But if I just use petsc~mpi without removing the others the build fails. This way I can just do spack install py-petsc4py~mpi
Instead I will need spack install py-petsc4py petsc~mpi~hdf5~hypre~superlu-dist

If I currently do spack install py-petsc4py ^petsc~mpi
In petsc build I get:
--==> 72186: petsc: Executing phase: 'install'
==> Error: RuntimeError: incompatible variants given
	hdf5 support requires "+mpi" to be activated
	hypre support requires "+mpi" to be activated
	superlu-dist support requires "+mpi" to be activated

@balay should petsc turn these off with ~ mpi.

Reply:
I think it would be good for petsc to add conflict statements for those packages but I'm happy to go with `depends_on('petsc~mpi', when='~mpi')`